components/connector/internal/tokens/service.go

```diff
@@ -42,22 +42,26 @@ func NewTokenService(cli GraphQLClient) *tokenService {
 	return &tokenService{cli: cli}
 }
 
-func (svc *tokenService) GetToken(ctx context.Context, clientId, consumerType string) (string, apperrors.AppError) {
-	token, err := svc.getOneTimeToken(ctx, clientId, consumerType)
+func (svc *tokenService) GetToken(ctx context.Context, clientId, authToken, consumerType string) (string, apperrors.AppError) {
+	token, err := svc.getOneTimeToken(ctx, clientId, authToken, consumerType)
 	if err != nil {
 		return "", apperrors.Internal("could not get one time token: %s", err)
 	}
 
 	return token, nil
 }
 
-func (svc *tokenService) getOneTimeToken(ctx context.Context, id, consumerType string) (string, error) {
+func (svc *tokenService) getOneTimeToken(ctx context.Context, id, authToken, consumerType string) (string, error) {
 	query, found := consumerTypeToQuery[consumerType]
 	if !found {
 		return "", errors.Errorf("No token generation for consumer type %s", consumerType)
 	}
 	req := gcli.NewRequest(fmt.Sprintf(query, "", id))
 
+	if authToken != "" {
```

Comment:
Why is this now neccesary? It seems to me that even with the previous implementation this internal token was needed. Isn't this done by the serviceAccountTokenTransport used for the httpClient which is used in the director client?

Reply:
Yes, you're totally right. Nice catch, strong eyes 👀 .